From 8896e5bc2a09fbaabaa9bc2ce6a393b681bd7c12 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 28 Apr 2019 22:27:58 +0300
Subject: [PATCH] [SW-0] pll: use API for CPU cycles

Instead of playing the include game in Makefile, use API function for
receiving the device specific CPU cycles information.

Note: we still play the include game, but start the preparation
for removing it.

Change-Id: Icb21b85a5a2d9075cdba13b2f17e9b57f582e5c7
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index a4f1dba..d415c53 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -37,7 +37,7 @@
 	if (bval == HL_PLL_REF_CLK || bval == HL_PLL_DIV_REF)
 		cnt = BYPASS_CYCLES_PER_USEC;
 	else
-		cnt = CPU_CYCLES_PER_USEC;
+		cnt = hl_pll_get_cpu_cycles_per_usec();
 
 	cnt = (cnt * usec) / MIN_CYCLES_PER_ITER;
 
diff --git a/common/pll.h b/common/pll.h
index 0f65ea3..ccfa22b 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -27,7 +27,6 @@
 
 #define SYS_REF_CLOCK		50000000ULL
 #define USEC_PER_SEC		1000000
-#define CPU_CYCLES_PER_USEC	(HL_CPU_CYCLES_PER_USEC / USEC_PER_SEC)
 #define BYPASS_CYCLES_PER_USEC	(SYS_REF_CLOCK / USEC_PER_SEC)
 #define MIN_CYCLES_PER_ITER	5
 
@@ -60,6 +59,7 @@
 
 int hl_pll_init(unsigned int *pll_partial_init);
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg);
+u64 hl_pll_get_cpu_cycles_per_usec(void);
 int hl_pll_partial_init_get(void);
 void hl_pll_partial_init_set(void);
 
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index 87a55bb..5a25a24 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -37,6 +37,11 @@
 	return mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
 }
 
+u64 hl_pll_get_cpu_cycles_per_usec(void)
+{
+	return 120000000ULL / USEC_PER_SEC;
+}
+
 static struct hl_pll_cfg gaudi_pll_cfg[] = {
 		{
 			.name = "CPU", .base = mmPSOC_CPU_PLL_BASE,
diff --git a/gaudi/gaudi.h b/gaudi/gaudi.h
index 7e2e70a..8522606 100644
--- a/gaudi/gaudi.h
+++ b/gaudi/gaudi.h
@@ -30,8 +30,6 @@
 #define mmNIC2_TPC_PLL_BASE		0xD7A000
 #define CPU_PLL_BASE			mmPSOC_CPU_PLL_BASE
 
-#define HL_CPU_CYCLES_PER_USEC		120000000
-
 #define PCI_DIV2_CLOCK			(830000000 / 5)
 #define UART_BAUDDIV			(PCI_DIV2_CLOCK / 115200)
 #define SPI_BAUDR			(PCI_DIV2_CLOCK / 25000000)
diff --git a/goya/goya.c b/goya/goya.c
index a61e384..9e9c5a9 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -163,6 +163,11 @@
 	return mmPSOC_GLOBAL_CONF_WARM_REBOOT;
 }
 
+u64 hl_pll_get_cpu_cycles_per_usec(void)
+{
+	return 133000000ULL / USEC_PER_SEC;
+}
+
 static struct hl_pll_cfg goya_pll_cfg[] = {
 	{
 		.name = "CPU", .base = mmCPU_PLL_BASE,
diff --git a/goya/goya.h b/goya/goya.h
index ac4ff5a..06d36f0 100644
--- a/goya/goya.h
+++ b/goya/goya.h
@@ -12,6 +12,4 @@
 #define mmIC_PLL_BASE		0x4A3000
 #define CPU_PLL_BASE		mmCPU_PLL_BASE
 
-#define HL_CPU_CYCLES_PER_USEC	133000000
-
 #define NON_RST_FLOPS_3		mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3
